gateway/webhook/webhook_test.go

```diff
@@ -281,3 +283,72 @@ func TestWebhookRequestHandlerWithOutputToGatewayAndSource(t *testing.T) {
 	assert.Equal(t, sampleJson, strings.TrimSpace(w.Body.String()))
 	_ = webhookHandler.Shutdown()
 }
+
+func TestRecordSourceTransformationErrors(t *testing.T) {
+	initWebhook()
+	ctrl := gomock.NewController(t)
+	mockGW := mockWebhook.NewMockGatewayI(ctrl)
+	statsStore := memstats.New()
+	webhookHandler := Setup(mockGW, statsStore)
+	reqs := []*webhookT{
+		{writeKey: "w1"}, {writeKey: "w2"}, {writeKey: "w1"}, {writeKey: "w3"}, {writeKey: "w2"}, {writeKey: "w1"},
+	}
+	mockGW.EXPECT().NewSourceStat(gomock.Any(), gomock.Any()).DoAndReturn(func(writeKey, reqType string) *gwStats.SourceStat {
+		if writeKey == "w1" {
+			return &gwStats.SourceStat{
+				Source:      "source1",
+				SourceID:    "sourceID1",
+				WriteKey:    writeKey,
+				ReqType:     reqType,
+				WorkspaceID: "workspaceID1",
+				SourceType:  "webhook1",
+			}
+		} else if writeKey == "w2" {
+			return &gwStats.SourceStat{
+				Source:      "source2",
+				SourceID:    "sourceID2",
+				WriteKey:    writeKey,
+				ReqType:     reqType,
+				WorkspaceID: "workspaceID2",
+				SourceType:  "webhook2",
+			}
+		} else if writeKey == "w3" {
+			return &gwStats.SourceStat{
+				Source:      "source3",
+				SourceID:    "sourceID3",
+				WriteKey:    writeKey,
+				ReqType:     reqType,
+				WorkspaceID: "workspaceID3",
+				SourceType:  "webhook3",
+			}
+		}
```

Comment:
[minor] if you feel it helps with readability
```suggestion
        switch writeKey {
        case "w1":
			return &gwStats.SourceStat{
				Source:      "source1",
				SourceID:    "sourceID1",
				WriteKey:    writeKey,
				ReqType:     reqType,
				WorkspaceID: "workspaceID1",
				SourceType:  "webhook1",
			}
        case "w2":
			return &gwStats.SourceStat{
				Source:      "source2",
				SourceID:    "sourceID2",
				WriteKey:    writeKey,
				ReqType:     reqType,
				WorkspaceID: "workspaceID2",
				SourceType:  "webhook2",
			}
        case "w3":
			return &gwStats.SourceStat{
				Source:      "source3",
				SourceID:    "sourceID3",
				WriteKey:    writeKey,
				ReqType:     reqType,
				WorkspaceID: "workspaceID3",
				SourceType:  "webhook3",
			}
		}
```